src/components/radioSelect.tsx

```diff
@@ -30,6 +32,7 @@ const RadioSelect: React.FC<RadioSelectProps> = ({
               name="single-select"
               type="radio"
               onChange={(event) => handleChange(event)}
+              disabled={isDisabled}
```

Comment:
I believe this disabled property can easily be extended for the `singleCheck` component. However, fix some of the requested changes in this branch first by following the previous issues from the other 2 older PRs.